### PR TITLE
Replace regex crate with glob crate 

### PR DESCRIPTION
**Pull request by lalitb**
## Changes

replace regex crate with glob crate to fix CI failure due to msrv issue:

``
error: package `regex-syntax v0.8.1` cannot be built because it requires rustc 1.65 or newer, while the currently active rustc version is 1.64.0 Error: Process completed with exit code 101.
``
## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-rust/blob/main/CONTRIBUTING.md) guidelines followed
* [x] Unit tests added/updated (if applicable)
* [x] Appropriate `CHANGELOG.md` files updated for non-trivial, user-facing changes
* [x] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: opentelemetry-sdk/src/metrics/view.rs [hunk 1]
```diff
@@ -177,3 +172,37 @@ pub fn new_view(criteria: Instrument, mask: Stream) -> Result<Box<dyn View>> {
...
+    use crate::metrics::Instrument;
+    use std::error::Error;
+    use std::result::Result;
+    #[test]
+    fn test_new_view() -> Result<(), Box<dyn Error>> {
+        let criteria = Instrument::new().name("counter");
+        let mask = Stream::new().aggregation(Aggregation::Sum);
+        let view = new_view(criteria, mask)?;
+
+        assert!(view
```
Comment (cijothomas) [inline opentelemetry-sdk/src/metrics/view.rs:189]: nit: sort into matching vs non-matching for easier readability.

**Review by cijothomas — APPROVED** [review]

**Comment by djc** [comment]
What does the spec have to say about what kind of matching should be supported here? It looks like the glob crate is barely being maintained, so unless our use case is directly related to its stated goal of "support for matching file paths against Unix shell style patterns", maybe it makes more sense to just bump the MSRV?

**Comment by lalitb** [comment]
> What does the spec have to say about what kind of matching should be supported here?

[specs](https://github.com/open-telemetry/opentelemetry-specification/blob/v1.26.0/specification/metrics/sdk.md#instrument-selection-criteria) needs matching of ‘?’ and ‘*’ metachars, so glob match should work fine.

**Comment by lalitb** [comment]
@djc whether we bump the MSRV now or later, this PR is still valid as fix for #1262. The name pattern containing ‘*’ and ‘?’ chars is more suited for glob matching, and so no need to convert it to regex.

**Review by TommyCpp — APPROVED** [review]
Given the use of the create is simple I think it's fine for now

**Review by djc — APPROVED** [review]

**Comment by lalitb** [comment]
The CI tests for failing for disk space getting full, not related to this PR:

rust_ci_error_1

I will try to cleanup some of the unused large packages ( dotnet/mono, android etc.) before running the tests if it helps. Else, we may have to split our tests to run on separate runners, or request for [large runner](https://docs.github.com/en/actions/using-github-hosted-runners/about-larger-runners/about-larger-runners)  / [self-hosted runners](https://github.com/open-telemetry/community/issues/1662#issuecomment-1729519261) with high disk space.  Any other suggestions are welcome.

**Comment by lalitb** [comment]
Ok, tests are successful now. Adding below cleanup recovers 18gb, and so providing total 34gb which seems to be enough for tests for now:

```yaml
    steps:
    - name: Free disk space
      run: |
        df -h
        sudo rm -rf /usr/local/lib/android
        sudo rm -rf /usr/share/dotnet
        df -h
```

Before cleanup:
```
Filesystem      Size  Used Avail Use% Mounted on
/dev/root        84G   66G   18G  80% /
tmpfs           3.4G  172K  3.4G   1% /dev/shm
tmpfs           1.4G  1.1M  1.4G   1% /run
tmpfs           5.0M     0  5.0M   0% /run/lock
/dev/sdb15      [10](https://github.com/open-telemetry/opentelemetry-rust/actions/runs/6537399729/job/17751156422?pr=1301#step:2:11)5M  6.1M   99M   6% /boot/efi
/dev/sda1        14G  4.1G  9.0G  31% /mnt
tmpfs           693M   [12](https://github.com/open-telemetry/opentelemetry-rust/actions/runs/6537399729/job/17751156422?pr=1301#step:2:13)K  693M   1% /run/user/1001
```

After cleanup:
```
Filesystem      Size  Used Avail Use% Mounted on
/dev/root        84G   50G   34G  60% /
tmpfs           3.4G  172K  3.4G   1% /dev/shm
tmpfs           1.4G  1.1M  1.4G   1% /run
tmpfs           5.0M     0  5.0M   0% /run/lock
/dev/sdb15      105M  6.1M   99M   6% /boot/efi
/dev/sda1        [14](https://github.com/open-telemetry/opentelemetry-rust/actions/runs/6537399729/job/17751156422?pr=1301#step:2:15)G  4.1G  9.0G  31% /mnt
tmpfs           693M   12K  693M   1% /run/user/1001
```

**Pull request merged**